Packs/DigitalGuardian/Integrations/DigitalGuardianARCEventCollector/DigitalGuardianARCEventCollector.py

```diff
@@ -0,0 +1,303 @@
+    Raises exceptions if something goes wrong.
+    Args:
+        client (Client): HelloWorld client to use.
+        params (Dict): Integration parameters.
+        first_fetch_time (int): The first fetch time as configured in the integration params.
+    Returns:
+        str: 'ok' if test passed, anything else will raise an exception and will fail the test.
+    """
+
+    try:
+        limit = int(params.get('Number_of_events', 1000))
+
+        fetch_events(
+            client=client,
+            last_run={},
+            first_fetch_time=first_fetch_time,
+            limit=limit,
+        )
+
+    except Exception as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def get_raw_events(client, days):
+    """
+       helper function that is used in get-events and fetch-events to get the actual events and sort them
+       Args:
+           client (Client): DG client
+           days (int): number of days to fetch
+       Returns:
+           list: list of events
+    """
+    outcome = []
+    event_list = []
+    token = client.get_token()
+    events = client.get_events(
+        days=days,
+        token=token
+    )
+    for item in events["fields"]:
+        outcome.append(item['name'])
+    for item in events["data"]:
+        result = dict(zip(outcome, item))
+        event_list.append(result)
+    event_list.sort(key=lambda item: (item["inc_mtime"], item["dg_guid"]))
+    return event_list
+
+
+def get_events(client, args):
+    """
+        Implement the get_events command
+        Args:
+            client (Client): DG client
+            args (dict): Command arguments
+        Returns:
+            list: list of events
+            commandresults: readable output
+    """
+    days = args.get("days")
+    event_list = get_raw_events(client, days)
+    limit = int(args.get("limit")) if args.get("limit") else None
+    if limit:
+        event_list = event_list[:limit]
+    hr = tableToMarkdown(name='Test Event', t=event_list)
+    return event_list, CommandResults(readable_output=hr)
+
+
+def create_events_for_push(event_list, last_time, id_list, limit):
+    """
+       Create events for pushing them and prepares the values for next_run save
+       Args:
+           event_list (list): list of events
+           last_time (str): time of last event from previous run
+           id_list (list): list of id's ingested
+           limit (int): max_fetch
+       Returns:
+           list: list of events
+           last_time: updates time of last event
+           id_list: list of id's
+    """
+    index = 0
+    event_list_for_push = []
+    for event in event_list:
+        if last_time:
+            if event.get("inc_mtime") < last_time:
+                continue
+            if event.get("dg_guid") in id_list:
+                continue
+            if last_time[:10] == event.get("inc_mtime")[:10]:
```

Comment:
Why are we only taking the first 10 digits

Reply:
the value in last time or inc_mtime is:
"2023-05-23 06:56:39"
so when I want to compare only the date part of current event (inc_mtime) and the time stored in last_run, I'm taking only the date part

Reply:
A better practice is to convert to datetime and extract date:
`str(myDateTimeObj.date())`

Reply:
Fixed as you suggested